javalib/src/main/scala/java/lang/String.scala

```diff
@@ -1,42 +1,812 @@
+      false
+    } else {
+      regionMatches(0, new _String(0, size, sb.getValue), 0, size)
+    }
+  }
+
+  def contentEquals(cs: CharSequence): scala.Boolean = {
+    val len = cs.length
+    if (len != count) {
+      false
+    } else if (len == 0 && count == 0) {
+      true
+    } else {
+      regionMatches(0, _String.valueOf(cs.toString), 0, len)
+    }
+  }
+
+  def matches(expr: _String): scala.Boolean =
+    Pattern.matches(expr, this)
+
+  def replaceAll(expr: _String, substitute: _String): _String =
+    Pattern.compile(expr).matcher(this).replaceAll(substitute)
+
+  def replaceFirst(expr: _String, substitute: _String): _String =
+    Pattern.compile(expr).matcher(this).replaceFirst(substitute)
+
+  def split(expr: _String): Array[String] =
+    Pattern.compile(expr).split(this)
+
+  def split(expr: _String, max: Int): Array[String] =
+    Pattern.compile(expr).split(this, max)
+
+  def subSequence(start: Int, end: Int): CharSequence =
+    subString(start, end)
+
+  def codePointAt(index: Int): Int =
+    if (index < 0 || index >= count) {
+      throw new IndexOutOfBoundsException()
+    } else {
+      Character.codePointAt(value, index + offset, offset + count)
+    }
+
+  def codePointBefore(index: Int): Int =
+    if (index < 1 || index > count) {
+      throw new IndexOutOfBoundsException()
+    } else {
+      Character.codePointBefore(value, index + offset)
+    }
+
+  def codePointCount(beginIndex: Int, endIndex: Int): Int =
+    if (beginIndex < 0 || endIndex > count || beginIndex > endIndex) {
+      throw new IndexOutOfBoundsException()
+    } else {
+      Character.codePointCount(
+          value, beginIndex + offset, endIndex - beginIndex)
+    }
+
+  def contains(cs: CharSequence): scala.Boolean =
+    indexOf(_String.valueOf(cs.toString)) >= 0
+
+  def offsetByCodePoints(index: Int, codePointOffset: Int): Int = {
+    val s = index + offset
+    val r =
+      Character.offsetByCodePoints(value, offset, count, s, codePointOffset)
+    r - offset
+  }
+
+  def getValue(): Array[Char] = value
 }
 
 object _String {
-  def valueOf(obj: Object): _String = ???
-  def valueOf(data: Array[Char]): _String = ???
-  def valueOf(data: Array[Char], offset: scala.Int, count: scala.Int): _String = ???
-  def valueOf(b: scala.Boolean): _String = ???
-  def valueOf(c: scala.Char): _String = ???
-  def valueOf(s: scala.Short): _String = ???
-  def valueOf(i: scala.Int): _String = ???
-  def valueOf(l: scala.Long): _String = ???
-  def valueOf(f: scala.Float): _String = ???
-  def valueOf(d: scala.Double): _String = ???
-  def format(format: String, args: scala.Array[Object]): _String = ???
+  final val CASE_INSENSITIVE_ORDER: Comparator[_String] =
+    new CaseInsensitiveComparator()
+  private final val ascii = {
+    val ascii = new Array[Char](128)
+    var i = 0
+    while (i < ascii.length) {
+      ascii(i) = i.toChar
+      i += 1
+    }
+    ascii
+  }
+
+  class ConsolePrintStream(out: java.io.OutputStream)
```

Comment:
This class was ported from harmony https://github.com/apache/harmony/blob/724deb045a85b722c961d8b5a83ac7a697319441/classlib/modules/luni/src/main/java/java/lang/String.java#L62